backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/QueryLoggerConf.java

```diff
@@ -0,0 +1,136 @@
+        Optional<PerHostPercentileTracker> percentileTracker,
+        Optional<Double> slowQueryLatencyThresholdPercentile,
+        Optional<Integer> maxLoggedParameters,
+        Optional<Integer> maxParameterValueLength,
+        Optional<Integer> maxQueryStringLength
+    ) {
+        this.constantThreshold = constantThreshold;
+        this.percentileTracker = percentileTracker;
+        this.slowQueryLatencyThresholdPercentile = slowQueryLatencyThresholdPercentile;
+        this.maxLoggedParameters = maxLoggedParameters;
+        this.maxParameterValueLength = maxParameterValueLength;
+        this.maxQueryStringLength = maxQueryStringLength;
+    }
+
+    public QueryLogger getQueryLogger(Cluster cluster) {
+        QueryLogger.Builder builder = QueryLogger.builder(cluster);
+
+        percentileTracker.map(percentileTracker ->
+            slowQueryLatencyThresholdPercentile.map(slowQueryLatencyThresholdPercentile ->
+                builder.withDynamicThreshold(percentileTracker, slowQueryLatencyThresholdPercentile)
+            )
+        );
+
+        constantThreshold.map(builder::withConstantThreshold);
+        maxLoggedParameters.map(builder::withMaxLoggedParameters);
+        maxParameterValueLength.map(builder::withMaxParameterValueLength);
+        maxQueryStringLength.map(builder::withMaxQueryStringLength);
+
+        return builder.build();
+    }
+
+    public static class QueryLoggerConfBuilder {
+        private Optional<Long> constantThreshold;
+        private Optional<PerHostPercentileTracker> percentileTracker;
+        private Optional<Double> slowQueryLatencyThresholdPercentile;
+        private Optional<Integer> maxLoggedParameters;
+        private Optional<Integer> maxParameterValueLength;
+        private Optional<Integer> maxQueryStringLength;
+
+        private QueryLoggerConfBuilder() {
+            constantThreshold = Optional.empty();
+            percentileTracker = Optional.empty();
+            slowQueryLatencyThresholdPercentile = Optional.empty();
+            maxLoggedParameters = Optional.empty();
+            maxParameterValueLength = Optional.empty();
+            maxQueryStringLength = Optional.empty();
+        }
+
+        public QueryLoggerConfBuilder withConstantThreshold(long constantThreshold) {
+            this.constantThreshold = Optional.of(constantThreshold);
+
+            return this;
+        }
+
+        public QueryLoggerConfBuilder withDynamicThreshold(PerHostPercentileTracker percentileTracker,
+                                                           double slowQueryLatencyThresholdPercentile) {
+            this.percentileTracker = Optional.of(percentileTracker);
+            this.slowQueryLatencyThresholdPercentile = Optional.of(slowQueryLatencyThresholdPercentile);
+
+            return this;
+        }
+
+        public QueryLoggerConfBuilder withMaxLoggedParameters(int maxLoggedParameters) {
+            this.maxLoggedParameters = Optional.of(maxLoggedParameters);
+
```

Comment:
I don't see benefits other than having more lines insertions in Github stats, but OK if you like it.